src/auth_v3.js

```diff
@@ -149,12 +149,25 @@ export function getClient (cozy, clientParams) {
   if (!cli.isRegistered()) {
     return Promise.reject(new Error('Client not registered'))
   }
-  return cozyFetchJSON(cozy, 'GET', `/auth/register/${cli.clientID}`, null, {
-    manualAuthCredentials: {
-      token: cli
-    }
-  })
-    .then((data) => createClient(data, cli))
+  return cozyFetchJSON(cozy, 'GET', `/auth/register/${cli.clientID}`, null,
+    {
+      manualAuthCredentials: {
+        token: cli
+      }
+    })
+    .catch((err) => {
+      // If we fall into an error while fetching the client (because of a
+      // bad connectivity for instance), we do not bail the whole process
+      // since the client should be able to continue with the persisted
+      // client and token.
+      //
+      // If it is an explicit Unauthorized error though, we bail, clear th
+      // cache and retry.
+      if (FetchError.isUnauthorized(err) || FetchError.isNotFound(err)) {
```

Comment:
`cozyFetchJSON` is used for many things and a 404 error does not mean that the client has been revoked in most cases